### PR TITLE
Use get_info to set valid_signals

### PR DESCRIPTION
**Pull request by araignee**
I think this is the way we want to set `TorControlProtocol.valid_signals`, but I'm not sure how to test it. I looked at [the test](https://github.com/meejah/txtorcon/blob/e41f9486db1a088ee7b79f914d56d6c372d914e6/test/test_torcontrolprotocol.py#L356-L379) where `TorControlProtocol.version` is set, since it's set in a similar way. But it does not seem like the test actually requests info from the Tor network? Is testing `valid_signals` like this good enough?

### CODE REVIEW

**Comment by meejah** [comment]
For testing: yes, that's good enough. There's a unit-test failing already that you'll have to fix, too which shows how this works:

https://github.com/meejah/txtorcon/blob/e41f9486db1a088ee7b79f914d56d6c372d914e6/test/test_torinfo.py#L54

It basically goes through and does the "tor side" of the bootstrap() stuff, so we test the TorControlProtocol parsers and things but don't actually have to connect to a "real" tor process.

So, you'll have to add an answer for "signals/names" at the right spot (e.g. give it a list of one or two dummy ones, and assert later that those are the only signals allowed.

Actually, it's kind of odd more of the unit-tests don't fail...

There are a couple end-to-end functional tests that run tor inside a Docker container, but that's overkill for this.

File: txtorcon/torcontrolprotocol.py [hunk 1]
```diff
@@ -716,14 +716,8 @@ def _bootstrap(self, *args):
...
-        # tor like the events...so this was taken from some version
-        # of the control-spec and must be kept up-to-date (or accpet
-        # any signal name and just wait for the reply?
-        # FIXME XXX there is now "GETINFO signal/names"
-        self.valid_signals = ["RELOAD", "DUMP", "DEBUG", "NEWNYM",
-                              "CLEARDNSCACHE"]
-
+        self.valid_signals = yield self.get_info('signal/names')
+	self.valid_signals = self.valid_signals['signal/names']
         self.version = yield self.get_info('version')
```
Comment (meejah) [inline txtorcon/torcontrolprotocol.py]: it occurs to me we should probably handle an error on ``GETINFO signal/names`` and set it to the fixed list from before in that case -- or at least figure out what tor version *doesn't* have it and whether backwards-compatibility really matters here.

**Comment by araignee** [comment]
@meejah Ok, would a `try/except` block catch an exception from one of these deferred objects?

**Comment by meejah** [comment]
yes, with @inlineCallbacks and yield's errors come as exceptions from the yield

**Comment by sammyshj** [comment]
We can also make use of a stem here, `Signal` which is actually `stem.util.enum.Enum` object
We can do something like:
`from stem import Signal`
`Signal.NEWNYM`
This is to issue a NEWNYM signal. @meejah Should I go ahead and create a PR to make use of this for the `optional-stem` branch?

**Comment by meejah** [comment]
I guess we could use the stem thing for the fallback list, but for consistency I think I'd rather keep txtorcon's original fallback list and use dynamic-discovery (i.e. ``GETINFO signal/names`` as this ticket is implementing) instead of a static list going forward.

**Comment by meejah** [comment]
What we really need to know here is what version of Tor introduced "getinfo signal/names" and what the list of valid signals is for tors prior to that. And whether its even worth supporting anything that *doesn't* have signal/names (i.e. maybe it's fine to just be an error if the tor doesn't have signals/names).

**Comment by meejah** [comment]
...or, as per the original comment, if there's no signal/names GETINFO we could just have an empty list of valid signals and accept anything (and if it's not allowed, tor will ultimately reply with a 500 to the "SIGNAL"). This actually might be the nicest way to do it, since then we can't ever be out-of-sync with tor's notion of valid signals.

**Comment by sammyshj** [comment]
@araignee Are you still working on this issue?

**Pull request closed without merging**